stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2495,38 +2482,62 @@ collections = [
 ]
 }%
 
+% for (Self, a_self, TypeParametersDecl, TypeParameters, AnyTypeParameters, Sequence, AnySequenceType) in collections:
+
 /// Header part of the native storage.
-internal struct _HashedContainerStorageHeader {
+internal struct _${Self}StorageHeader<${TypeParameters}> {
+
+%if Self == 'Set':
+  internal typealias Key = ${TypeParameters}
+  internal typealias Value = ${TypeParameters}
+%end
+
   internal init(capacity: Int) {
     self.capacity = capacity
   }
 
-  internal var capacity: Int
+  internal let capacity: Int
   internal var count: Int = 0
-  internal var maxLoadFactorInverse: Double =
+  internal let maxLoadFactorInverse: Double =
     _hashContainerDefaultMaxLoadFactorInverse
-}
 
-% for (Self, a_self, TypeParametersDecl, TypeParameters, AnyTypeParameters, Sequence, AnySequenceType) in collections:
+  // Placeholder values until the struct can be initialized.
+  internal var initializedEntries: _UnsafeBitMap! = nil
+  internal var keys: UnsafeMutablePointer<Key>! = nil
+% if Self == 'Dictionary':
+  internal var values: UnsafeMutablePointer<Value>! = nil
+% end
+}
 
 /// An instance of this class has all `${Self}` data tail-allocated.
 /// Enough bytes are allocated to hold the bitmap for marking valid entries,
 /// keys, and values. The data layout starts with the bitmap, followed by the
 /// keys, followed by the values.
 @_versioned
-final internal class _Native${Self}StorageImpl<${TypeParameters}> {
+final internal class _Native${Self}Storage<${TypeParameters}> :
+  ManagedBuffer<_${Self}StorageHeader<${TypeParameters}>, UInt8> {
   // Note: It is intended that ${TypeParameters}
   // (without : Hashable) is used here - this storage must work
   // with non-Hashable types.
 
-  internal typealias StorageImpl = _Native${Self}StorageImpl
+  internal typealias StorageHeader = _${Self}StorageHeader<${TypeParameters}>
+  internal typealias BufferPointer =
+    ManagedBufferPointer<StorageHeader, UInt8>
+  internal typealias Storage = _Native${Self}Storage<${TypeParameters}>
 
 %if Self == 'Set': # Set needs these to keep signatures simple.
   internal typealias Key = ${TypeParameters}
+  internal typealias Value = ${TypeParameters}
+  internal typealias SequenceElementWithoutLabels = Element
+%else:
+  internal typealias SequenceElementWithoutLabels = (Key, Value)
 %end
 
-  internal var _body: _HashedContainerStorageHeader
+  internal var _body: StorageHeader
 
+  /// The number of _logical elements_ the ${a_self} can store. Not to be
+  /// confused with `capacity`, the total number of raw `UInt8` bytes stored
```

Comment:
This (now deleted) comment was struggling around the inherited capacity from ManagedBuffer; now that we don't inherit from them, this is no longer a thing. (yay!)